callbackify/index.ts

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as is from 'is';
+
+export interface CallbackifyAllOptions extends CallbackifyOptions {
+  /**
+   * Array of methods to ignore when callbackifying.
+   */
+  exclude?: string[];
+}
+
+export interface CallbackifyOptions {
+  /**
+   * Pass to the callback a single arg instead of an array.
+   */
+  singular?: boolean;
+}
+
+export interface CallbackMethod extends Function {
+  callbackified_?: boolean;
+}
+
+/**
+ * Wraps a promisy type function to conditionally call a callback function.
+ *
+ * @param {function} originalMethod - The method to callbackify.
+ * @param {object=} options - Callback options.
+ * @param {boolean} options.singular - Pass to the callback a single arg instead of an array.
+ * @return {function} wrapped
+ */
+export function callbackify(originalMethod, options?: CallbackifyOptions) {
```

Comment:
I think we'll want this is a standalone package to use in all the clients instead of having it live in the client repo.

Reply:
Yes, of course. This is here only for test/show purposes. Once agreed I will submit PR as `googleapis/nodejs-callbackify`